maas/client/viscera/machines.py

```diff
@@ -190,18 +195,70 @@ class Machine(Object, metaclass=MachineType):
         if comment is not None:
             params["comment"] = comment
         data = await self._handler.deploy(**params)
-        return type(self)(data)
+        if wait is None:
+            return type(self)(data)
+        else:
+            # Wait for the machine to be fully deployed
+            machine = type(self)(data)
+            while machine.status == NodeStatus.DEPLOYING:
+                await asyncio.sleep(wait)
+                data = await self._handler.read(system_id=self.system_id)
+                machine = type(self)(data)
+            return machine
```

Comment:
You should check here to make sure its now DEPLOYED. It its not then you should raise an Exception.

`raise FailedDeployment(machine.error, obj=machine)`

We have not created exceptions like this before. But it would be good to include a reason why it failed (I believe this is returned in the `error` property on the machine), and also the whole object.